src/integrationTest/java/uk/gov/hmcts/reform/wacaseeventhandler/MessageReadinessTest.java

```diff
@@ -0,0 +1,126 @@
+import java.util.List;
+import java.util.concurrent.ThreadLocalRandom;
+import java.util.stream.Stream;
+
+import static java.util.concurrent.TimeUnit.MILLISECONDS;
+import static java.util.concurrent.TimeUnit.SECONDS;
+import static org.awaitility.Awaitility.await;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.when;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.put;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.content;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+
+@SpringBootTest
+@AutoConfigureMockMvc(addFilters = false)
+@ActiveProfiles("db")
+@TestPropertySource(properties = {"azure.servicebus.enableASB=true",
+        "azure.servicebus.connection-string="
+                + "Endpoint=sb://REPLACE_ME/;SharedAccessKeyName=REPLACE_ME;SharedAccessKey=REPLACE_ME",
+        "azure.servicebus.topic-name=test",
+        "azure.servicebus.subscription-name=test",
+        "azure.servicebus.ccd-case-events-subscription-name=test",
+        "azure.servicebus.retry-duration=2"})
+public class MessageReadinessTest {
+
+    private static final boolean DLQ_EMPTY = true;
+    private static final boolean DLQ_NOT_EMPTY = false;
+
+    @MockBean
+    private DeadLetterQueuePeekService deadLetterQueuePeekService;
+
+    @MockBean
+    private LaunchDarklyFeatureFlagProvider launchDarklyFeatureFlagProvider;
+
+    @Autowired
+    private MockMvc mockMvc;
+
+    private String randomMessageId() {
+        return "" + ThreadLocalRandom.current().nextLong(1000000);
+    }
+
+    private static String getCaseEventMessage() {
+        return "{\n"
+                + "  \"EventInstanceId\" : \"some event instance Id\",\n"
+                + "  \"EventTimeStamp\" : \"" + LocalDateTime.now() + "\",\n"
+                + "  \"CaseId\" : \"caseIdValue\",\n"
+                + "  \"JurisdictionId\" : \"ia\",\n"
+                + "  \"CaseTypeId\" : \"asylum\",\n"
+                + "  \"EventId\" : \"some event Id\",\n"
+                + "  \"NewStateId\" : \"some new state Id\",\n"
+                + "  \"UserId\" : \"process_true\",\n"
+                + "  \"MessageProperties\" : {\n"
+                + "      \"property1\" : \"test1\"\n"
+                + "  }\n"
+                + "}";
+    }
+
+    private MvcResult postMessage(String messageId) throws Exception {
+        return mockMvc.perform(put("/messages/" + messageId + "?from_dlq=false")
+                .contentType(MediaType.APPLICATION_JSON)
+                .content(getCaseEventMessage()))
+                .andExpect(status().isCreated())
+                .andReturn();
+    }
+
+    private static Stream<Arguments> provideDlqStateAndExpectedMessageState() {
+        return Stream.of(
+                Arguments.of(DLQ_EMPTY, MessageState.READY),
```

Comment:
Just trying to clarify;
Should this be DLQ_EMPTY, MessageState.NEW?

As per AC in the ticket?

Reply:
It's the expected state. This is resolved. 👍